src/CircularArrays.jl

```diff
@@ -65,4 +65,14 @@ function Base.showarg(io::IO, arr::CircularArray, toplevel)
     # toplevel && print(io, " with eltype ", eltype(arr))
 end
 
+function Base.deleteat!(a::CircularVector, inds)
+    deleteat!(a.data, (unique ∘ sort ∘ map)(i -> mod(i, eachindex(a.data)), inds))
```

Comment:
This sadly fails if `inds` is a `Tuple` since `sort` isn't implemented for Tuples. It should be `sort ∘ unique ∘ map`, probably.

Reply:
In fact, if we do this, then I don't think `sort` will be necessary anymore, since any `inds` should be sorted when we're given it, and `unique` will preserve that order.

Reply:
I further managed to improve performance and reduce the number of necessary allocations by changing this line to be more consistent with the rest
```suggestion
    deleteat!(a.data, sort(unique(mod1.(inds, length(a.data)))))
```

Reply:
I further managed to slightly improve performance by making this line more consistent with the rest.
```suggestion
    deleteat!(a.data, sort(unique(mod1.(inds, length(a.data)))))
```

Reply:
Mhm, that's unfortunate that `sort` isn't supported for `Tuple`. We could just handle `Tuple` separately I suppose.

Using `unique` before `sort` isn't much better because we'd end up allocating an array...
```julia
julia> unique((1,2,3))
3-element Vector{Int64}:
 1
 2
 3
```

Reply:
What if we do `sort! ∘ unique`? If `unique` always returns a mutable array, this should have the same number of allocations as before.

Reply:
Good point!